From 5c642d5c41313b1839e0ce96d7f6c20ba87d0c35 Mon Sep 17 00:00:00 2001
From: Hai-Ying Wang <Haiying.Wang@freescale.com>
Date: Thu, 4 Oct 2012 21:05:16 +0000
Subject: [PATCH 085/227] fsl_bman: change the buffer pool amount to 64 for
 all BMan 2.1 SoC

The latest BMan block guide shows all SoCs with BMan 2.1 have 64 buffer pools.

Signed-off-by: Haiying Wang <Haiying.Wang@freescale.com>
Signed-off-by: Andy Fleming <afleming@freescale.com>
[Kevin: Original patch taken from fsl sdk 1.3.1
QorIQ-SDK-V1.3.1-SOURCE-20121220-yocto.iso.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/staging/fsl_qbman/bman_config.c |   19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/drivers/staging/fsl_qbman/bman_config.c b/drivers/staging/fsl_qbman/bman_config.c
index 6f3fa0c..875dbad 100644
--- a/drivers/staging/fsl_qbman/bman_config.c
+++ b/drivers/staging/fsl_qbman/bman_config.c
@@ -210,12 +210,6 @@ static void bm_get_version(struct bman *bm, u16 *id, u8 *major, u8 *minor)
 	*minor = v & 0xff;
 }
 
-static void bm_get_version_2(struct bman *bm, u8 *ip_cfg)
-{
-	u32 v = bm_in(IP_REV_2);
-	*ip_cfg = v & 0xff;
-}
-
 static u32 __generate_thresh(u32 val, int roundup)
 {
 	u32 e = 0;	/* co-efficient, exponent */
@@ -364,7 +358,6 @@ static int __init fsl_bman_init(struct device_node *node)
 	int ret, standby = 0;
 	u16 id;
 	u8 major, minor;
-	u8 ip_cfg;
 
 	ret = of_address_to_resource(node, 0, &res);
 	if (ret) {
@@ -397,17 +390,7 @@ static int __init fsl_bman_init(struct device_node *node)
 		bman_pool_max = 8;
 	} else if ((major == 2) && (minor == 1)) {
 		bman_ip_rev = BMAN_REV21;
-		bm_get_version_2(bm, &ip_cfg);
-		switch (ip_cfg) {
-			case 0:
-			case 1:
-				bman_pool_max = 64;
-				break;
-			case 2:
-				bman_pool_max = 32;
-			default:
-				break;
-		}
+		bman_pool_max = 64;
 	} else {
 		pr_warning("unknown Bman version, default to rev1.0\n");
 	}
-- 
1.7.9.7

